runtime/near-vm-runner/src/cache.rs

```diff
@@ -292,3 +292,61 @@ pub mod wasmer1_cache {
         }
     }
 }
+
+pub(crate) fn precompile_contract_impl(
+    vm_kind: VMKind,
+    wasm_code: &ContractCode,
+    config: &VMConfig,
```

Comment:
I think this is OK for the initial implementation, but I am not entirely happy that we are passing the whole `VMConfig` here. In partticular, we hash `ExtCosts` into the key, but we don't actually use them during compilation. I think we might want to have

```rust
struct VMConfig {
    compile: VMCompileConfig,
    run: VMRunConfig,
}
```

and pass only the `Compile` part here. But this definitely can, and should be done separately.